src/runtime/arc.ts

```diff
@@ -558,18 +557,13 @@ ${this.activeRecipe.toString()}`;
           // TODO(shans): clean this up when we have interfaces for Singleton, Collection, etc.
           // tslint:disable-next-line: no-any
           await (newStore as any).set(particleClone);
-        } else if (recipeHandle.fate === 'copy') {
-          const copiedStoreRef = this.findStoreById(recipeHandle.id);
-          let copiedStore: StorageProviderBase;
-          if (copiedStoreRef instanceof StorageStub) {
-            copiedStore = await copiedStoreRef.inflate();
-          } else {
-            copiedStore = copiedStoreRef;
-          }
+        } else if (['copy', 'map'].includes(recipeHandle.fate)) {
+          const copiedStoreRef = this.context.findStoreById(recipeHandle.id) as StorageStub;
+          const copiedStore = await copiedStoreRef.inflate(this.storageProviderFactory);
```

Comment:
i'm passing the arc's storageProviderFactory to the inflate method here, because otherwise the copied store is inflated in the context's storageProviderFactory, having speculative execution affect the real arc's context (because the same context object is used for both). 
i don't think this is the right way to do it though, WDYT?

Reply:
discussed offline and agreed that this approach is ok for now.